### PR TITLE
Begin with network admin compatibility

### LINKED ISSUE #29: Multisite Compatibility
Right now it doesn't matter if you activate the plugin for the whole network or for each site individually.

However, if you network activate it, it's currently not possible to disable widgets in the network admin dashboard. Shouldn't be too hard to allow this.

### PR DESCRIPTION
**Pull request by swissspidy**
With this patch, the network admin looks like this:

screen shot 2018-05-02 at 20 26 04

To do:

* [x] Show message after saving settings.  
`add_settings_error()` doesn't work because there's a redirect happening. We could add a GET parameter in that case, .e.g. `&updated=<count>`.

Fixes #29.

### CODE REVIEW

File: classes/class-wp-widget-disable.php [hunk 1]
```diff
@@ -119,6 +123,17 @@ protected function get_path() {
 		return plugin_dir_path( __DIR__ );
 	}
 
+	/**
+	 * Returns the plugin basename.
+	 *
+	 * @since 1.8.0
```
Comment (swissspidy) [inline classes/class-wp-widget-disable.php]: Needs to be 1.9.0.

File: classes/class-wp-widget-disable.php [hunk 2]
```diff
@@ -161,6 +189,29 @@ public function settings_errors() {
 		settings_errors( 'wp-widget-disable' );
 	}
 
+	/**
+	 * Saves network options in the network admin.
+	 *
+	 * @since 1.8.0
```
Comment (swissspidy) [inline classes/class-wp-widget-disable.php]: Needs to be 1.9.0.

File: views/admin.php [hunk 3]
```diff
@@ -32,11 +32,21 @@
 if ( isset( $_GET['tab'] ) && 'dashboard' === $_GET['tab'] ) {
```
Comment (ocean90) [inline views/admin.php]: Looks like this needs to be
`if ( is_network_admin() || ( isset( $_GET['tab'] ) && 'dashboard' === $_GET['tab'] ) ) {`
because it's currently displaying the sidebar widgets.

Comment (swissspidy) [reply views/admin.php]: Right. I accidentally removed that in the last merge commit.

**Comment by ocean90** [comment]
When the site option doesn't exist yet you get the update message twice, same as in https://github.com/wearerequired/WP-Widget-Disable/issues/11. 

Should `set_default_options()` set the default site option if `is_multisite()`? Since that's only called by `register_activation_hook()` what about plugin updates?

**Comment by swissspidy** [comment]
Yeah, we could totally extend `set_default_options()` and perhaps always call it on `init`. However, I'm not a huge fan of just adding an empty options anyway.

I'd much rather like to investigate why this happens.

For instance, we call `settings_errors( 'wp-widget-disable' );` on `admin_notices`. That should be called on multisite as well, so there's no need to call `settings_errors()` in `views/admin.php` for multisite

In `save_network_options()` I used `general` instead of `wp-widget-disable`, which is also incorrect:

https://github.com/wearerequired/WP-Widget-Disable/blob/d584d0f218986f77e6a4b2306a4f61c969173d78/classes/class-wp-widget-disable.php#L213-L215

tl;dr:

* Remove `settings_errors()` call from view
* Use correct setting when using `get_settings_errors()` and `add_settings_error()`
* Ideally remove necessity for `set_default_options()`

**Comment by ocean90** [comment]
@swissspidy Changed a few things but did you see my mention of #11 and #18 which is basically https://core.trac.wordpress.org/ticket/21989?

**Comment by ocean90** [comment]
`admin_footer_text()` and `print_admin_styles()` are currently ignored on the network admin screen because `$screen->base` has `-network` as a suffix which `add_submenu_page()` does not add. For `print_admin_styles()` see #34, do we still want `admin_footer_text()`?

**Comment by swissspidy** [comment]
Even though we could just add the suffix for that case, I don‘t think there‘s much value in it really.
